### DISCUSSION TITLE
Earthworks v2.1 release tasks

### DISCUSSION THREAD

**Discussion by supreethms1809** [EarthWorksOrg/EarthWorks#26] [General]
- [x] Add Perlmutter machine to config_machines.xml. Create a branch off of develop, add my changes and create a PR. (@supreethms1809 ) : [git@github.com:EarthWorksOrg/ccs_config_cesm.git](git@github.com:EarthWorksOrg/ccs_config_cesm.git)

- [x] We will be getting MPAS7.x openacc version (not the 8.x version of mpas) for the V2.1 release. (@gdicker1 )

- [ ] MPAS 7.x in CAM performance numbers. CPU vs GPU numbers.(@supreethms1809 ) (**low priority)

- [x] Changes to ccs_config to include the following options --ngpus-per-node $GPUS_PER_NODE --gpu-type $GPU_TYPE --gpu-offload $GPU_OFFLOAD" (@gdicker1  to confirm)

- [x]  Jim/Jian's changes to include OpenACC flags via Depends.nvhpc. This includes changes to cime and ccs_config (@gdicker1  to add). (@supreethms1809) to test if we need to list all the MPAS files in Depends.nvhpc or if we can just carry the OpenACC flags to the MPAS Makefile in CAM.

- [ ]  Move the relavent files needed for run to Perlmutter using Dylan's directory. (@supreethms1809)

- [ ]  Test the EW compsets for various resolutions (@supreethms1809 ) on Derecho and Perlmutter.

- [x] What is the CAM tag version that we will be using (@gdicker1) - CAM 145 tag + aerosol patch + Compatibility changes needed for MPAS 7.x openacc to work inside CAM. 

- [x] Creating release tags for all the relevant repositories. (@gdicker1)
- [x] EW version 2.1 release notes (@areanddee)

**Comment by supreethms1809** [comment]
@sherimickelson @areanddee @gdicker1  Here are the release tasks pending. Please feel free to comment on this.

**Comment by gdicker1** [comment]
Post Release Team Meeting Feb 12th (new items):

- [x] (@dazlich) Check FullyCoupled CPU on Derecho with OneAPI compiler (ensure ifx works)
- [x] Meet 9am Feb 14th to do final release discussion

**Reply by dazlich** [reply]
Have verified the coupled model runs and restarts for ifx. I tested 120km, and 15km/58layers. The gnu compiler also works for these configurations, nvhpc runs but fails to restart as expected.

**Comment by areanddee** [comment]
Don-

Great, thanks for doing this. Good to know it works.

On another note, I’m thinking ahead to future goals. Have you ever tried
7.5 km ocean/sea ice or even fully coupled? What would be entailed in such
setting up for such a test?

regards,

Rich

**Comment by dazlich** [comment]
I have done a 7.5 km data atm/ mpas ocean-seaice run. I would love to test the 7.5km fully coupled model if I were provided the topo, init, and land files

Sent from my iPhone

On Feb 13, 2024, at 8:53 AM, areanddee ***@***.***> wrote:

﻿

** Caution: EXTERNAL Sender **

Don-

Great, thanks for doing this. Good to know it works.

On another note, I’m thinking ahead to future goals. Have you ever tried
7.5 km ocean/sea ice or even fully coupled? What would be entailed in such
setting up for such a test?

regards,

Rich

**Comment by gdicker1** [comment]
Work that remains after [v2.1 Release](https://github.com/EarthWorksOrg/EarthWorks/releases/tag/release-ew2.1):
- Resolve any bugs that come from the release (esp. FullyCoupled runs)
- Move the relevant files needed for run to Perlmutter
    - It would especially be nice to have this in a common area. Supreeth and I tried before the release to use personal directories and couldn't resolve some file permission issues that would enable the other to use the same directory.
- MPAS 7.x in CAM performance numbers. CPU vs GPU numbers
- Test the EW compsets for various resolutions (@supreethms1809 ) on Derecho and Perlmutter

Further discussion related to the release should go to: #30
